src/main/blackbox/blackbox.c

```diff
@@ -1097,6 +1142,12 @@ static void loadMainState(timeUs_t currentTimeUs)
         blackboxCurrent->motor[i] = lrintf(motor[i]);
     }
 
+#ifdef USE_DSHOT_TELEMETRY
+    for (int i = 0; i < motorCount; i++) {
+        blackboxCurrent->rpm[i] = getDshotTelemetry(i);
```

Comment:
Just now saw that this logged eRPM and not actual RPM
```suggestion
        blackboxCurrent->rpm[i] = erpmToRpm(getDshotTelemetry(i));
```

Reply:
> Just now saw that this logged eRPM and not actual RPM

That's intentional (for now at least). Logging eRPM saves a little bit of calculation on the FC and might avoid some loss of precision. The eRPM to RPM calculation is done in Black box explorer.

tl;dr for the stream of thought bellow: the exact value to be logged is still TBD.

I still haven't decided on exactly which value that should be logged. The getDshotTelemetry function raises a few alarm bells when I look at it, especially when extended dhsot telemetry might come into play. There are too much calculations and too many side effects there for my liking right now, and I'm not sure that it won't interfere with the RPM based gyro filters.
It could also be a good idea to filter the RPM signal before logging to help the compression, but then it might as well log the value used by the RPM filters. I'll have to look at how much information there is left on higher frequencies. In the very short log I have now there is just noise.